thunderfish/psdtype.py

```diff
@@ -0,0 +1,173 @@
+import numpy as np
+import sys
+
+def bin_it(power, freq_bins, max_freq, res):
+    """
+    Take a 1-D array of powers (from powerspectrums), transforms it into dB and devides it into several bins.
+
+    :param power:           (1-D array) power array of a psd.
+    :param freq_bins:       (float) width of frequency bins in which the psd shall be divided (Hz).
+    :param max_freq:        (float) maximum frequency that shall be provided in the separated power array.
+    :param res:             (float) the frequency resolution of the power array comming from a psd.
+                            res = freqs[-1]/len(freqs)
+    :return power_db:       (2-D array) list of lists with the in bins divided powers of the powerspectrum transforemed
+                            into dB.
+    """
+    power_db = []
+    for trial in np.arange(max_freq / freq_bins):
+        tmp_power_db = 10.0 * np.log10(power[trial * int( freq_bins / (res) ) : (trial +1) * int( freq_bins / (res) ) - 1])
+        power_db.append(tmp_power_db)
+    return power_db
+
+def get_bin_percentiles(power_db):
+    """
+    Takes a 2-D array if lists. For every list the function calculates several percentails that are later returned.
+
+    :param power_db:        (2-D array) list of lists with the in bins divided powers of the powerspectrum transforemed
+                            into dB.
+    :return:                (2-D array) for every bin four values are calulated and stored in separate lists. These four
+                            values are percentiles of the respective bins.
+    """
+    power_db_top = np.ones(len(power_db))
+    power_db_upper_middle = np.ones(len(power_db))
+    power_db_lower_middle = np.ones(len(power_db))
+    power_db_bottom = np.ones(len(power_db))
+
+    for fbin in np.arange(len(power_db)):
+        power_db_top[fbin] = np.percentile(power_db[fbin], 99)
+        power_db_upper_middle[fbin] = np.percentile(power_db[fbin],75)
+        power_db_lower_middle[fbin] = np.percentile(power_db[fbin], 25)
+        power_db_bottom[fbin] = np.percentile(power_db[fbin], 1)
+
+    return [power_db_top, power_db_upper_middle, power_db_lower_middle, power_db_bottom]
+
+def get_example_data(audio_file, channel=0, verbose=None):
+    """
+    This function shows in part the same components of the thunderfish.py poject. Here several moduls are used to load
+    some data to dispay the functionality of the psdtype.py modul.
+
+    :param audio_file:      (string) filepath of a audiofile that shall be used for the analysis.
+    :return power:          (1-D array) power array of a psd.
+    :return freqs:          (1-D array) frequency array of a psd.
+    """
+    cfg = ct.get_config_dict()
+
+    if verbose is not None:  # ToDo: Need to document the whole cfg-dict thing.
+        cfg['verboseLevel'][0] = verbose
+    channel = channel
+
+    # load data using dataloader module
```

Comment:
the rest of the function can go into the main script (if you want, also as a function)